### DISCUSSION TITLE
Getting 'wrong' metadata

### DISCUSSION THREAD

**Discussion by neal3000** [python-pillow/Pillow#8647] [Q&A] [answered]
I have some code that uses pillow to read in the metadata added by Comfyui to a PNG file. And that works fine.
But if I take an image and use imagemagick to convert it, keeping the meta data, and then try and read it I get metatadata:{'chromaticity': (0.3127, 0.329, 0.64, 0.33, 0.3, 0.6, 0.15, 0.06)} which is not what I want.
Using exiftool -PNG:all  ComfyUI_04572_.png. from the command line I see the metadata I want
Datetimestamp                   : 2024-12-31T02:55:02+00:00
Prompt                          : {"5": {"inputs": {"width": 1440, "height": 1280, "batch_size": 1},....

I'm guessing the issue is there is multiple metadata in the PNG file, and the "chromaticity" is taking precedence. If that is the case, is there any way to tell Pillow to look for alternative metadata, or return all possible metadata ?

Thank

ps I have tried alternatives, and none of them have worked (pypng and piexif)

**Comment by radarhere** [comment]
Could you attach a copy of the ComfyUI image, and a copy of the ImageMagick image?

Ideally, it would also be helpful if you could provide a simple example of Pillow code reading the metadata from the ComfyUI image.

**Comment by neal3000** [comment]
my code:
```python
image = PILImage.open(filename)
metadata = image.info
```
```
  "data": {
    "chromaticity": [
      "0.3127",
      "0.329",
      "0.64",
      "0.33",
      "0.3",
      "0.6",
      "0.15",
      "0.06"
    ]
  }
}
```
exiftool
```
exiftool -PNG:all ComfyUI_04525_.png 
Image Width                     : 1440
Image Height                    : 1280
Bit Depth                       : 8
Color Type                      : RGB
Compression                     : Deflate/Inflate
Filter                          : Adaptive
Interlace                       : Noninterlaced
White Point X                   : 0.3127
White Point Y                   : 0.329
Red X                           : 0.64
Red Y                           : 0.33
Green X                         : 0.3
Green Y                         : 0.6
Blue X                          : 0.15
Blue Y                          : 0.06
Background Color                : 255 255 255
Datecreate                      : 2024-12-30T07:05:02+00:00
Datemodify                      : 2024-12-30T07:05:02+00:00
Datetimestamp                   : 2024-12-31T02:53:56+00:00
Prompt                          : {"5": {"inputs": {"width": 1440, "height": 1280, "batch_size": 1}, "class_type": "EmptyLatentImage"}, "8": {"inputs": {"samples": ["15", 0], "vae": ["14", 0]}, "class_type": "VAEDecode"}, "9": {"inputs": {"filename_prefix": "ComfyUI", "images": ["8", 0]}, ....
```
ComfyUI_04525_ - https://github.com/user-attachments/assets/fd340680-7a10-4621-b455-efc12056a2ca

**Reply by neal3000** [reply]
ps the code:
```python
from PIL import Image as PILImage
image = PILImage.open(filename)
metadata = image.info
```
works fine if the image came from Comfyui and had not been resized by imagemagick

**Comment by radarhere** [comment, marked as answer]
```python
from PIL import Image
im = Image.open("comfy.png")
im.load()
for k, v in im.info.items():
    print(k+": "+str(v)[:100])
```
will give you
```
chromaticity: (0.3127, 0.329, 0.64, 0.33, 0.3, 0.6, 0.15, 0.06)
date:create: 2024-12-30T07:05:02+00:00
date:modify: 2024-12-30T07:05:02+00:00
date:timestamp: 2024-12-31T02:53:56+00:00
prompt: {"5": {"inputs": {"width": 1440, "height": 1280, "batch_size": 1}, "class_type": "EmptyLatentImage"}
workflow: {"last_node_id": 28, "last_link_id": 54, "nodes": [{"id": 20, "type": "PreviewImage", "pos": {"0": 1
```

Is that all you were after?

**Reply by radarhere** [reply]
As an explanation, see https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html#png
> As of Pillow 6.0, EXIF data can be read from PNG images. However, unlike other image formats, EXIF data is not guaranteed to be present in [`info`](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.info) until [`load()`](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.load) has been called.

**Comment by neal3000** [comment]
Thank you so much, that's 100% what I was looking for.

simply using 
```python
image0 = PILImage.open(filename)
mtext = image0.text
```
worked
